crates/matrix-sdk-base/src/rooms/normal.rs

```diff
@@ -1622,4 +1704,117 @@ mod tests {
             Raw::from_json_string(json!({ "event_id": event_id }).to_string()).unwrap(),
         ))
     }
+
+    fn timestamp(minutes_ago: u32) -> MilliSecondsSinceUnixEpoch {
+        MilliSecondsSinceUnixEpoch::from_system_time(
+            SystemTime::now().sub(Duration::from_secs((60 * minutes_ago).into())),
+        )
+        .expect("date out of range")
+    }
+
+    fn call_member_state_event(
+        memberships: Vec<Membership>,
+        ev_id: &str,
+        user_id: &UserId,
+    ) -> AnySyncStateEvent {
+        let content = CallMemberEventContent::new(memberships);
+
+        AnySyncStateEvent::CallMember(SyncStateEvent::Original(OriginalSyncCallMemberEvent {
+            content,
+            event_id: OwnedEventId::from_str(ev_id).unwrap(),
+            sender: user_id.to_owned(),
+            // we can simply use now here since this will be dropped when using a MinimalStateEvent
+            // in the roomInfo
+            origin_server_ts: timestamp(0),
+            state_key: user_id.to_owned(),
+            unsigned: StateUnsigned::new(),
+        }))
+    }
+
+    fn membership_for_my_call(
+        device_id: &str,
+        membership_id: &str,
+        minutes_ago: u32,
+    ) -> Membership {
+        let application = Application::Call(CallApplicationContent::new(
+            "my_call_id_1".to_owned(),
+            ruma::events::call::member::CallScope::Room,
+        ));
+        let foci_active = vec![Focus::Livekit(LivekitFocus::new(
+            "my_call_foci_alias".to_owned(),
+            "https://lk.org".to_owned(),
+        ))];
+
+        assign!(
+            Membership::from(MembershipInit {
+                application,
+                device_id: device_id.to_owned(),
+                expires: Duration::from_millis(3_600_000),
+                foci_active,
+                membership_id: membership_id.to_owned(),
+            }),
+            { created_ts: Some(timestamp(minutes_ago)) }
+        )
+    }
+
+    fn receive_state_events(room: &Room, events: Vec<&AnySyncStateEvent>) {
+        let mut guard = room.inner.write();
+        ObservableWriteGuard::update(&mut guard, |g| {
+            for e in events {
+                g.handle_state_event(e);
+            }
+        });
```

Comment:
This will send out an update of the observable even `handle_state_event` returns `false` for all events. Should use `update_if`, also no need for a write guard.

```suggestion
        room.inner.update_if(|info| {
            events.into_iter().any(|ev| info.handle_state_event(ev))
        });
```